.github/workflows/ci.yml

```diff
@@ -20,110 +20,137 @@ jobs:
       fail-fast: false
       matrix:
         target: [
-          { "os": "ubuntu-latest",  "toolchain": "x86_64-unknown-linux-gnu", "name": "Linux GNU" },
-          { "os": "macOS-latest",   "toolchain": "x86_64-apple-darwin",      "name": "macOS" },
-          { "os": "windows-latest", "toolchain": "x86_64-pc-windows-msvc",   "name": "Windows MSVC" },
-          { "os": "windows-latest", "toolchain": "x86_64-pc-windows-gnu",    "name": "Windows GNU" }
+          { "os": "ubuntu-latest",  "toolchain": "x86_64-unknown-linux-gnu", "cargo_cmd": "cargo",      "name": "Linux GNU" },
+          { "os": "ubuntu-latest",  "toolchain": "x86_64-pc-windows-msvc",   "cargo_cmd": "cargo xwin", "name": "Linux - Windows MSVC" },
+          { "os": "macOS-latest",   "toolchain": "x86_64-apple-darwin",      "cargo_cmd": "cargo",      "name": "macOS" },
+          { "os": "windows-latest", "toolchain": "x86_64-pc-windows-msvc",   "cargo_cmd": "cargo",      "name": "Windows MSVC" },
+          { "os": "windows-latest", "toolchain": "x86_64-pc-windows-gnu",    "cargo_cmd": "cargo",      "name": "Windows GNU" }
         ]
-        channel: [stable, beta, nightly]
+        channel: [stable , beta, nightly]
+    env: 
+      XWIN_ARCH: "x86_64"
+      WINEDEBUG: "-all"
+      CARGO_BUILD_TARGET: ${{ matrix.target.toolchain }}
 
     steps:
     - uses: actions/checkout@v1
     - uses: actions-rs/toolchain@v1
+      if: ${{ matrix.target.name != 'Linux - Windows MSVC' }}
       with:
         toolchain: ${{ format('{0}-{1}', matrix.channel, matrix.target.toolchain) }}
         profile: minimal
         override: true
+    - uses: actions-rs/toolchain@v1
+      if: ${{ matrix.target.name == 'Linux - Windows MSVC' }}
+      with:
+        toolchain: ${{ matrix.channel }}
+        target: ${{ matrix.target.toolchain }}
+        profile: minimal
+        override: true
+
+    - name: install dependencies (Linux - Windows MSVC)
+      if: ${{ matrix.target.name == 'Linux - Windows MSVC' }}
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install --quiet -y clang-tools llvm lld 
+        sudo ln -s /usr/bin/clang-cl-[1-9][0-9] /usr/bin/clang-cl
+        sudo ln -s /usr/bin/lld /usr/bin/lld-link
+        sudo apt-get install --quiet --no-install-recommends -y wine-stable winetricks
+        winetricks nocrashdialog
+        cargo install cargo-xwin --target x86_64-unknown-linux-gnu
```

Comment:
Wow, thank you for figuring all this out. It's probably time for me to refactor all of these copy-pasted jobs into a Python script or something, so that additions like this aren't so painful in the future.